SU-SWS/stanford_components
# EARTH-000 Added 4 new twig functions to make things easier

# READY FOR REVIEW

# Summary
- Created 4 new twig functions to make things much better
1. `{{ link|openlink }}` - This will process the given string/render array and build an opening `<a>` tag with a title attribute to maintain accessibility. Also allows passing attributes. like `{{ link|openlink({class:'some-class-here', 'aria-value':'Some readable Value'}) }}`
2. `{{ link|closelink }}` - This will close the above open tag only if the open tag was created. 
3. `{{ field|dropcap }}` - This does the same thing we've been doing in a single filter. `strip_tags`, `trim`, then `substr(0,1)`
4. `{{ field|getimg }}` - gets just the image src url.

Both 1 & 2 help with accessibility and don't always require all the `{{ if link }}...{{ endif }}`. It'lll also allow us to use normal link fields with user entered title text.
3 will just help clean up the templates since we seem to use it alot and help make them all the same.
4 will help when adding an entity field to the region. it doesn't need to have the format as "url to image"

I edit the postcard pattern as an example.

# Needed By (Date)
- When you can

# Urgency
- Low critical

# Steps to Test

1. Checkout branch
2. `drush cr`
3. view pattners page & existing content that uses the patterns to verify things still look correct.
4. Review twig extension code.
